doc/user/content/sql/show-types.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "SHOW TYPES"
+description: "`SHOW TYPES` returns a list of the data types in your Materialize instance."
+menu:
+  main:
+    parent: 'sql'
+---
+
+`SHOW TYPES` returns a list of the data types in your Materialize instance. By default, only user-created columns are returned.
+
+## Syntax
+
+{{< diagram "show-types.svg" >}}
+
+Field | Use
+------|-----
+`EXTENDED` |  Returns system columns as well as user-created columns. By default, only user-created columns are returned.
+`FULL`| Returns the type of column (user or system).
+
+## Examples
+
+### Show custom data types
+
+```sql
+SHOW TYPES;
+```
+```
+      name
+----------------
+  int4_list
+  int4_list_list
+  int4_map
+  int5_map
```

Comment:
`int4_*` refers to the fact that the underlying scalar type of the object is an `int4`. It probably doesn't make sense to have `int5`! (Other `int` options include `int2` and `int8`.)